editoast/src/views/authz.rs

```diff
@@ -112,6 +113,40 @@ pub(in crate::views) async fn whoami(
     }))
 }
 
+#[editoast_derive::route]
+#[utoipa::path(
+    get,
+    path = "",
+    tag = "authz",
+    responses((
+        status = 200,
+        description = "Get the groups of the current user",
+        body = inline(Vec<Group>),
+    ))
+)]
+pub(in crate::views) async fn user_groups(
+    Extension(auth): AuthenticationExt,
+    State(AppState {
+        regulator, db_pool, ..
+    }): State<AppState>,
+) -> Result<Json<Vec<Group>>> {
+    let authorizer = auth.authorizer()?;
+    let user_id = authorizer.user_id();
+    let user_groups = regulator.user_groups(&authz::User(user_id)).await?;
+    let groups_id: Vec<i64> = user_groups.iter().map(|authz::Group(id)| *id).collect();
+
+    let (result, missing_ids) =
+        editoast_models::Group::retrieve_batch(&mut db_pool.get().await?, groups_id).await?;
+
+    if !missing_ids.is_empty() {
+        tracing::warn!(missing_count = missing_ids.len(),
```

Comment:
```suggestion
        tracing::error!(missing_count = missing_ids.len(),
```

That's an error. At no point the groups defined in OpenFGA should differ from what we know in the DB.

Reply:
@leovalais I think it's possible since deleting a group will only remove it from our PostgreSQL DB and not from fga. Even with a GC, this could happen, and we should only be warned.

Reply:
I don't think we have anything to delete groups currently though. (cf. [CLI](https://github.com/OpenRailAssociation/osrd/blob/dev/editoast/src/client/group.rs)). We settled on the need for a GC because of cascading deletions only. We don't want to have to remove FGA tuples of scenarios when we delete a project, even though PG cascades automatically.

But this doesn't apply to groups however. Now we can change and decide to never eagerly clean OpenFGA when possible and let the GC perform all deletions, in which case agreed it's not an error (then maybe not even a warning I'd argue).

Reply:
> Now we can change and decide to never eagerly clean OpenFGA when possible and let the GC perform all deletions.

It's what I imagined in the first place. Do we see flaws in this strategy? IMO it seems simpler since all fga tuples have the same policy (no deletion when the source object is deleted and then the GC handles their deletion).

> [!NOTE]
> I actually had to delete some users/groups. I did it directly in pg since we don't have a cli. I did not make a deletion on fga side :grimacing:  

If you agree with this strategy, let's close this discussion; otherwise, we can have a small discussion.

@woshilapin you might have an opinion on this subject.

Reply:
I don't have a strong opinion on that. Either way, GC or not, is probably fine. But I'd say `warn` is probably enough in any case, since it won't affect the behavior of the application (even if it is a situation in which we should not be in the case of a solution without GC).

Reply:
> It's what I imagined in the first place. Do we see flaws in this strategy?

Oh ok, I'm fine with this as well.

> let's close this discussion

Sure I didn't intent the last comment to be blocking.